File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/People.java
package dev.radley.omgstarwars.models;


import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * People model represents an individual person or character within the Star Wars universe.
 */
public class People extends SWModel implements Serializable {

    public String gender;
    public String height;
    public String mass;

    @SerializedName("birth_year")
    public String birthYear;

    @SerializedName("hair_color")
    public String hairColor;

    @SerializedName("homeworld")
    public String homeWorldUrl;

    @SerializedName("skin_color")
    public String skinColor;

    @SerializedName("films")
    public ArrayList<String> filmsUrls;

    @SerializedName("species")
    public ArrayList<String> speciesUrls;

    @SerializedName("starships")
    public ArrayList<String> starshipsUrls;

    @SerializedName("vehicles")
    public ArrayList<String> vehiclesUrls;

    @Override
    public ArrayList<String> getRelatedFilms() {
        return filmsUrls;
    }

    @Override
    public ArrayList<String> getRelatedSpecies() {
        return speciesUrls;
    }

    @Override
    public ArrayList<String> getRelatedStarships() {
        return starshipsUrls;
    }

    @Override
    public ArrayList<String> getRelatedVehicles() {
        return vehiclesUrls;
    }
}

File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/People.kt
package dev.radley.omgstarwars.models


import com.google.gson.annotations.SerializedName

import java.io.Serializable
import java.util.ArrayList

/**
 * People model represents an individual person or character within the Star Wars universe.
 */
class People : SWModel(), Serializable {

    var gender: String = ""
    var height: String = ""
    var mass: String = ""

    @SerializedName("birth_year")
    var birthYear: String = ""

    @SerializedName("hair_color")
    var hairColor: String = ""

    @SerializedName("homeworld")
    var homeWorldUrl: String = ""

    @SerializedName("skin_color")
    var skinColor: String = ""

    @SerializedName("films")
    override var relatedFilms: ArrayList<String>? = null

    @SerializedName("species")
    override var relatedSpecies: ArrayList<String>? = null

    @SerializedName("starships")
    override var relatedStarships: ArrayList<String>? = null

    @SerializedName("vehicles")
    override var relatedVehicles: ArrayList<String>? = null
}
